### DISCUSSION TITLE
Directly use sdkjs in client application

### DISCUSSION THREAD

**Discussion by YousefED** [ONLYOFFICE/DocumentServer#1697] [Q&A]
Hi,

I'm trying to create / load an excel file and access cells / formules directly from client-side javascript code. However, at this moment I do not want to show the entire OnlyOffice interface.

I'm able to build `cells/sdk-all.js`, but I don't think embedding this in my application is enough to get access to `AscCommonExcel` and related functions. Do you have any guidance on how I can build / embed the project and get access to the underlying API?

Thanks,

**Comment by ShockwaveNN** [comment]
I think DocumentServer's `sdkjs` is not designed in that way, to be able to hide the interface

But we got another product - DocumentBuilder 
It's [standalone product](https://api.onlyoffice.com/docbuilder/basic) and also [a part of DocumentServer](https://api.onlyoffice.com/editors/documentbuilderapi) - you can send requests to your DocumentServer with commands to modify files and got result of this modification

Not sure if it will cover your user case, but I don't think accessing `sdk-all.js` methods directly is supported without any hacks

Alos I'll move this to discussions, since it's not an issue per se

**Reply by YousefED** [reply]
Thanks. DocumentBuilder seems to be a server-side (c++) library. I'm want to use sdkjs (or something similar) on the client side, for example to be able to evaluate Excel formulas, or render specific parts of a spreadsheet (such as a chart), but not necessarily in a full-fledged "excel like" interface

**Reply by ShockwaveNN** [reply]
[DocumentBuilder](https://www.onlyoffice.com/download.aspx#builder) can be also a client-side application. Just a standalone application

I think you can use it to evaluate Excel formulas, not sure about rendering only parts of spreadsheet - you can convert spreadsheet to png and cut specific parts by coordinates via some third-party library

**Reply by YousefED** [reply]
But it doesnt run in the browser right? (Sorry, Inmeant browser-based instead of client side)

**Reply by ShockwaveNN** [reply]
It doesn't require browser at all

If you need browser based solution I don't think I can help you

**Reply by ShockwaveNN** [reply]
Actually you can write a JS wrapper around DocumentBuilder web interface

Send API request to it's page and getting result and handling them in your JS code

**Reply by YousefED** [reply]
Thanks for your quick replies! 

Yes, of course its possible to do via an API and run documentbuilder serverside. However, Im looking for a browser based solution (no backend). For this I’d need to figure out a way to use sdkjs right? (So that brings us back to my original question)

**Reply by ShockwaveNN** [reply]
I don't think our code can be easily be adapted to usage without backed, it was never designed such way and there is a lot of dependencies in `web-apps` and backed part, I don't think it will be achievable without rewrite of major part of codebase
